iati/core/rulesets.py

```diff
@@ -261,39 +331,41 @@ def is_valid_for(self, dataset):
 
         Raises:
             AttributeError: When an argument is given that does not have the required attributes.
-            XPathEvalError(lxml.etree.XPathEvalError): When no valid XPath is available.
 
         """
-        path_queries = set()
-        found_paths = set()
+        context_elements = self._find_context_elements(dataset)
 
-        for path in self.paths:
-            path_queries.add(path)
-            if dataset.xml_tree.xpath(path) != list():
-                found_paths.add(path)
+        for context_element in context_elements:
+            if self._evaluate_condition(context_element):
+                return None
```

Comment:
New problem with the Standard - unclear whether it should `continue` (ignore the Rule for this context and move onto the next one) or `return` (skip the entire thing).

Reply:
(I'd be inclined to `continue` due to the order of nesting within the JSON)

Reply:
the original code returns None as a skip equivalent but it also extracts context elements differently so it's unclear what should be done here. Will add a Todo to prompt discussion later.